packages/tkg-clusterclass/bundle/config/packageinstalls/tkg-infra-clusterclass-pi.yaml

```diff
@@ -72,3 +73,177 @@ metadata:
+                #! by a defaulting webhook instead for extended logic.
+                default: "v1.23"
+                description: "enforceVersion sets the version for the enforce PodSecurityConfiguration mode."
+              audit:
+                type: string
+                enum: ["", "privileged", "baseline", "restricted"]
+                description: "audit sets the level for the audit PodSecurityConfiguration mode. One of \"\", privileged, baseline, restricted."
+                nullable: true
+              auditVersion:
+                type: string
+                #! Version defaults to v1.23 for now. When v1.25 is part of tanzu-framework, the defaulting should be done
+                #! by a defaulting webhook instead for extended logic.
+                default: "v1.23"
+                description: "auditVersion sets the version for the audit PodSecurityConfiguration mode."
+              warn:
+                type: string
+                enum: ["", "privileged", "baseline", "restricted"]
+                description: "warn sets the level for the warn PodSecurityConfiguration mode. One of \"\", privileged, baseline, restricted."
+                nullable: true
+              warnVersion:
+                type: string
+                #! Version defaults to v1.23 for now. When v1.25 is part of tanzu-framework, the defaulting should be done
+                #! by a defaulting webhook instead for extended logic.
+                default: "v1.23"
+                description: "warnVersion sets the version for the warn PodSecurityConfiguration mode."
+              exemptions:
+                type: object
+                description: "exemption configuration for the PodSecurityConfiguration."
+                properties:
+                  namespaces:
+                    type: array
+                    items:
+                      type: string
+                    description: "namespaces excluded to apply PodSecurityConfiguration Admission."
+      #@overlay/match missing_ok=True
+      patches:
+      #@overlay/remove
+      #@overlay/match by=overlay.map_key("name"),missing_ok=True
+      - name: podSecurityStandard
+      #@overlay/append
+      - name: podSecurityStandard
+        description: "Adds an admission configuration for PodSecurity to the kube-apiserver."
+        definitions:
+        - selector:
+            apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+            kind: KubeadmControlPlaneTemplate
+            matchResources:
+              controlPlane: true
+          jsonPatches:
+          - op: add
+            path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraArgs/admission-control-config-file"
+            value: "/etc/kubernetes/kube-apiserver-admission-pss.yaml"
+          - op: add
+            path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraVolumes/-"
```

Comment:
has this been fixed? I remember having issue early on where adding with `-` wasn't properly adding to the end of the array. I just want to confirm that this has been tested to work.

Reply:
I tested it if it works, yes.

Adding with `-` requires to have an empty array first. This gets added by the `kcp-empty-array-overlay.yaml`.
Note: having the empty array in the to-be patched object (in this case `KubeadmControlPlaneTemplate`) does not always work because the field may be treated with omitEmpty. Thats why it is required to have the "add empty array" patch.

With Halifax and Runtime SDK we will be able to write better patches using the topology mutation hooks.